app/parse.py

```diff
@@ -8,9 +18,102 @@ class Quote:
     tags: list[str]
 
 
+@dataclass
+class Author:
+    biography: str
+
+
+def parse_single_quote(page_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=page_soup.select_one(".text").text,
+        author=page_soup.select_one(".author").text,
+        tags=[tag.text for tag in page_soup.select(".tag")]
+    )
+
+
+def parse_single_author(page_soup: BeautifulSoup) -> Author:
+    return Author(
+        biography=page_soup.select_one(
+            ".author-details"
+        ).text.replace("\n", " "),
+    )
+
+
+def write_quotes_in_file(quotes: [Quote], output_csv_path: str) -> None:
+    quote_fields = [field.name for field in fields(Quote)]
+    with open(output_csv_path, "w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(quote_fields)
+        writer.writerows([astuple(quote) for quote in quotes])
+
+
+def write_authors_in_file(authors: [Author]) -> None:
+    author_fields = [field.name for field in fields(Author)]
+    with open("authors.csv", "w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(author_fields)
+        writer.writerows([astuple(author) for author in authors])
+
+
+def get_authors_page_soup(
+        page_soup: list[BeautifulSoup]
+) -> list[BeautifulSoup]:
+    list_authors = []
+    for author_url in page_soup:
+        url = urljoin(
+            BASE_URL,
+            author_url.select_one("a").get("href")
+        )
+        if url not in AUTHORS_URL_SET:
+            page = requests.get(url).content
+            list_authors.append(BeautifulSoup(page, "html.parser"))
+        AUTHORS_URL_SET.add(url)
```

Comment:
Change constants???????
https://en.wikipedia.org/wiki/Constant_(computer_programming)

Reply:
This is a global variable that stores links to authors in set to remove duplication.
I did not find another way to do this without classes and with the implementation of parallelism (so as not to pull data from processes)

Reply:
It is constant. Then rename it to lowercase
https://intellipaat.com/blog/tutorial/python-tutorial/python-variables/#no3